### PR TITLE
rgw: resolve bug where marker was not advanced during garbage collection

### PR DESCRIPTION
**Pull request by ivancich**
This resolves a bug where no GC progress could be made under certain
rare circumstances. This also adds additional logging.

Fixes: http://tracker.ceph.com/issues/38408
Signed-off-by: J. Eric Ivancich <ivancich@redhat.com>

### CODE REVIEW

**Comment by ivancich** [comment]
jenkins retest

File: src/rgw/rgw_gc.cc [hunk 1]
```diff
@@ -254,11 +254,15 @@ class RGWGCIOManager {
       ++index;
     }
   }
-};
+}; // class RGWGCIOManger
```
Comment (mattbenjamin) [inline src/rgw/rgw_gc.cc]: I find these helpful

**Comment by ivancich** [comment]
retest this please

**Comment by smithfarm** [comment]
This is pretty bizarre, but AFAICT the commit in this PR is a subset of the changes in #26601 - i.e. this one could have been closed instead of merged, and there would be no difference in the resulting code.

**Pull request merged**